### DISCUSSION TITLE
Update a model where relation id is equal to

### DISCUSSION THREAD

**Discussion by KasparRosin** [prisma/prisma#8297] [Q&A] [answered]
How does one update a model based on it's relation?

Schema:
```prisma
model Post {
   id Int
   user_id Int
   
   user User @relation(...)
}
```

Would like to do something like this:
```ts
prisma.post.update({
    where: {id: 1, user_id: 3}    
})
```

The compiled PostWhereUniqueInput looks like this: 
```ts
export type PostWhereUniqueInput = {
    id?: number
}
```

From reading docs, I understand that i can only achieve this by doing the following:
```ts
prisma.user.update({
    where: { id: 3 },
    data: {
       posts: {
           update: {
               where: { id: 1 },
           }
       }
    }
})
```

**Comment by ryands17** [comment, marked as answer]
@KasparRosin 👋 

You can use [updateMany](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#updatemany) to update records with multiple non-unique conditions.

`update` will only work for unique fields.

**Reply by KasparRosin** [reply]
~~user's id and post id are technically both unique, since only one user with an id exists on user table.~~

**Reply by KasparRosin** [reply]
Would it be possible to configure the BatchPayload to return an array of updated items, so i could return only the first one, instead of it returning the count of updated items? Or will i have to create a separate query to find post by id?

**Reply by ryands17** [reply]
Adding a `@unique` on the field should make this work in `update` then if you're sure that it's unique. So:

```prisma
model Post {
   id Int
   user_id Int @unique
   
   user User @relation(...)
}
```

As the above is a 1-1 relation, adding unique should work and then you should be able to see both of those fields in `update` after running `prisma generate`.

**Reply by ryands17** [reply]
> Would it be possible to configure the BatchPayload to return an array of updated items, so i could return only the first one, instead of it returning the count of updated items? Or will i have to create a separate query to find post by id?

A separate query would be needed.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
